onnx/backend/test/runner/__init__.py

```diff
@@ -288,28 +288,33 @@ def run(test_self, device):  # type: (Any, Text) -> None
                 self.assert_similar_outputs(ref_outputs, outputs,
                                             rtol=model_test.rtol,
                                             atol=model_test.atol)
-
             for test_data_dir in glob.glob(
                     os.path.join(model_dir, "test_data_set*")):
                 inputs = []
                 inputs_num = len(glob.glob(os.path.join(test_data_dir, 'input_*.pb')))
                 for i in range(inputs_num):
                     input_file = os.path.join(test_data_dir, 'input_{}.pb'.format(i))
-                    tensor = onnx.TensorProto()
-                    with open(input_file, 'rb') as f:
-                        tensor.ParseFromString(f.read())
-                    inputs.append(numpy_helper.to_array(tensor))
+                    self._load_proto(input_file, inputs, model.graph.input[i].type)
                 ref_outputs = []
                 ref_outputs_num = len(glob.glob(os.path.join(test_data_dir, 'output_*.pb')))
                 for i in range(ref_outputs_num):
                     output_file = os.path.join(test_data_dir, 'output_{}.pb'.format(i))
-                    tensor = onnx.TensorProto()
-                    with open(output_file, 'rb') as f:
-                        tensor.ParseFromString(f.read())
-                    ref_outputs.append(numpy_helper.to_array(tensor))
+                    self._load_proto(output_file, ref_outputs, model.graph.output[i].type)
                 outputs = list(prepared_model.run(inputs))
                 self.assert_similar_outputs(ref_outputs, outputs,
                                             rtol=model_test.rtol,
                                             atol=model_test.atol)
 
         self._add_test(kind + 'Model', model_test.name, run, model_marker)
+
+    def _load_proto(self, proto_filename, target_list, model_type_proto):  # type: (Text, List[Union[np.ndarray[Any], List[Any]]], TypeProto) -> None
+        with open(proto_filename, 'rb') as f:
+            protobuf_content = f.read()
+            if model_type_proto.HasField('sequence_type'):
+                sequence = onnx.SequenceProto()
+                sequence.ParseFromString(protobuf_content)
+                target_list.append(numpy_helper.to_list(sequence))
+            elif model_type_proto.HasField('tensor_type'):
```

Comment:
Do we want to leave a message in the else branch saying : loading proto of that specific type (Map/Sparse Tensor) is currently not supported ?

Reply:
Sounds good. Added. Thanks

Reply:
A comment is nice, but I was suggesting printing something for the user